projects/aca-content/src/lib/components/search/search-results/search-results.component.ts

```diff
@@ -189,17 +201,30 @@ export class SearchResultsComponent extends PageComponent implements OnInit {
     this.columns = this.extensions.documentListPresets.searchResults || [];
 
     if (this.route) {
-      this.route.params.forEach((params: Params) => {
-        // eslint-disable-next-line no-prototype-builtins
-        this.searchedWord = params.hasOwnProperty(this.queryParamName) ? params[this.queryParamName] : null;
-        if (this.searchedWord) {
-          this.queryBuilder.update();
+      this.route.queryParams.pipe(takeUntil(this.onDestroy$)).subscribe(async (params: Params) => {
+        this.encodedQuery = params[this.queryParamName] ? params[this.queryParamName] : null;
+        this.searchedWord = extractSearchedWordFromEncodedQuery(this.encodedQuery);
+        this.updateUserQuery();
+        const filtersFromEncodedQuery = extractFiltersFromEncodedQuery(this.encodedQuery);
+        if (filtersFromEncodedQuery !== null) {
+          const filtersToLoad = this.queryBuilder.categories.length;
+          let loadedFilters = this.searchedWord === '' ? 0 : 1;
+          this.queryBuilder.filterLoaded
+            .asObservable()
+            .pipe(takeUntil(this.onDestroy$), takeUntil(this.loadedFilters$))
+            .subscribe(async () => {
```

Comment:
Can we somehow avoid subscribe in other subscribe? Maybe we can use some rxjs operator?

Reply:
I'm not sure if we can in this scenario, as there is quite complex logic in this subscribe and this second subscribe is a side effect which is required in some circumstances. But I added some condition to make sure it's not creating infinite amount of subscribtions - each one is completed if new value comes in